apps/deploy-web/src/components/billing-usage/TrendIndicator/TrendIndicator.tsx

```diff
@@ -0,0 +1,64 @@
+import React from "react";
+import { isToday } from "date-fns";
+import { GraphDown, GraphUp } from "iconoir-react";
+
+const COMPONENTS = {
+  GraphUp,
+  GraphDown
+};
+
+type HistoryData = Array<{ date: string; [key: string]: number | string | null }>;
+
+type Keys<Data extends HistoryData> = keyof Omit<Data[number], "date">;
+
+export type TrendIndicatorProps<Field extends Keys<Data>, Data extends HistoryData = HistoryData> = {
+  isFetching: boolean;
+  data: Data;
+  field: Field;
+  components?: typeof COMPONENTS;
+};
+
+export const TrendIndicator = <Field extends string & Keys<Data>, Data extends HistoryData = HistoryData>({
+  isFetching,
+  data,
+  field,
+  components: { GraphUp, GraphDown } = COMPONENTS
+}: TrendIndicatorProps<Field, Data>) => {
+  const trendData = React.useMemo(() => {
+    if (data.length < 2) return null;
+
+    const sortedData = data
+      .filter(item => item[field] !== undefined && item[field] !== null)
+      .sort((a, b) => new Date(a.date).getTime() - new Date(b.date).getTime());
```

Comment:
Why sorting? Would this be expected behaviour on the client code? shouldn't data be sorted as needed by the client? i might just want to get trend providing 2 numbers in a reverse order, idk. anyway, our backend does the sorting via args provided by ui. not sure we need to sort again.